dotCMS/src/main/java/com/dotcms/vanityurl/business/VanityUrlAPIImpl.java

```diff
@@ -54,641 +46,238 @@
-                    );
-        }
+    if (contentlet instanceof VanityUrl) {
+      return (VanityUrl) contentlet;
     }
 
-    private User getUser(final Contentlet contentlet) {
-        User user;
-        try {
-            user = this.userAPI.loadUserById(contentlet.getModUser());
-        } catch (Exception e) {
-            Logger.debug(this,e.getMessage(),e);
-            user = this.systemUser;
-        }
-        return user;
+    final DefaultVanityUrl vanityUrl = new DefaultVanityUrl();
+    vanityUrl.setContentTypeId(contentlet.getContentTypeId());
+    try {
+      this.contentletAPI.copyProperties(vanityUrl, contentlet.getMap());
+    } catch (Exception e) {
+      throw new DotStateException("Vanity Url Copy Failed", e);
     }
+    vanityUrl.setHost(contentlet.getHost());
+    vanityUrl.setFolder(contentlet.getFolder());
+    CacheLocator.getContentletCache().add(vanityUrl);
+    return vanityUrl;
+  } // getVanityUrlFromContentlet.
 
-    /**
-     * Get the list of cached Vanity URLs associated to a given site and SYSTEM_HOST
-     *
-     * @param siteId The current site Id
-     * @param languageId The current language Id
-     * @param includeSystemHost True if we want to include in the result the SYSTEM_HOST cache
-     * contents
-     * @return A set of CachedVanityUrl, if null is because the cache needs to be initialized
-     */
-    private List<CachedVanityUrl> getVanityUrlBySiteAndLanguageFromCache(final String siteId,
-            final long languageId,
-            final Boolean includeSystemHost) {
-
-        //First search in cache with the given site id
-        List<CachedVanityUrl> foundVanities =
-                this.vanityUrlServices.getCachedVanityUrlList(siteId, languageId);
-
-        //null means we need to initialize the cache for this site
-        if (null != foundVanities &&
-                includeSystemHost &&
-                !siteId.equals(Host.SYSTEM_HOST)) {
-            //Now search in cache with the SYSTEM_HOST
-            final List<CachedVanityUrl> systemHostFoundVanities =
-                    this.vanityUrlServices.getCachedVanityUrlList(Host.SYSTEM_HOST, languageId);
-
-            if (null != systemHostFoundVanities) {
-                foundVanities = ImmutableList.<CachedVanityUrl>builder()
-                        .addAll(foundVanities)
-                        .addAll(systemHostFoundVanities)
-                        .build();
-            } else {
-                //This means we need to initialize the cache for the SYSTEM_HOST
-                foundVanities = null;
-            }
-        }
+  
+  @CloseDBIfOpened
+  @Override
+  public void validateVanityUrl(final Contentlet contentlet) {
 
-        return foundVanities;
-    } // getVanityUrlBySiteAndLanguageFromCache.
+    Language language = Try.of(()-> WebAPILocator.getLanguageWebAPI().getLanguage(HttpServletRequestThreadLocal.INSTANCE.getRequest())).getOrElse(APILocator.getLanguageAPI().getDefaultLanguage());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'language' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)